synthtool/languages/node_mono_repo.py

```diff
@@ -0,0 +1,397 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from jinja2 import FileSystemLoader, Environment
+from pathlib import Path
+import re
+from synthtool import _tracked_paths, gcp, shell, transforms
+from synthtool.gcp import samples, snippets
+from synthtool.log import logger
+from synthtool.sources import git
+from typing import Any, Dict, List, Optional, Callable
+import logging
+import shutil
+from synthtool.languages import common
+
+
+_REQUIRED_FIELDS = ["name", "repository", "engines"]
+_TOOLS_DIRECTORY = "/synthtool"
+_GENERATED_SAMPLES_DIRECTORY = "./samples/generated"
+
+
+def read_metadata(relative_dir: str):
+    """
+    read package name and repository in package.json from a Node library.
+
+    Returns:
+        data - package.json file as a dict.
+    """
+    with open(Path(relative_dir, "./package.json").resolve()) as f:
+        data = json.load(f)
+
+        if not all(key in data for key in _REQUIRED_FIELDS):
+            raise RuntimeError(
+                f"package.json is missing required fields {_REQUIRED_FIELDS}"
+            )
+
+        repo = git.parse_repo_url(data["repository"])
+
+        data["repository"] = f'{repo["owner"]}/{repo["name"]}'
+        data["repository_name"] = repo["name"]
+        data["lib_install_cmd"] = f'npm install {data["name"]}'
+        engines_field = re.search(r"([0-9][0-9])", data["engines"]["node"])
```

Comment:
I would switch this to `[0-9]+`, prevents a bug if Node ever goes to weekly releases like Chrome (which is up to version `101`).